src/Examine.AzureDirectory/AzureLuceneDirectory.cs

```diff
@@ -47,32 +46,31 @@ public class AzureLuceneDirectory : ExamineDirectory
         public AzureLuceneDirectory(
             string connectionString,
             string containerName,
-            Lucene.Net.Store.Directory cacheDirectory,
-            bool compressBlobs = false,
-            string rootFolder = null,
-            bool isReadOnly = false)
+            Lucene.Net.Store.Directory cacheDirectory,  bool compressBlobs = false,
+            string rootFolder = null)
         {
             if (connectionString == null) throw new ArgumentNullException(nameof(connectionString));
 
             if (string.IsNullOrWhiteSpace(containerName))
                 throw new ArgumentException("Value cannot be null or whitespace.", nameof(containerName));
-            IsReadOnly = isReadOnly;
             _storageAccountConnectionString = connectionString;
             CacheDirectory = cacheDirectory;
+            _helper = new AzureHelper();
```

Comment:
Newing up dependencies in a constructor is an anti pattern. I don't think refactoring out into AzureHelper helps with making the code more reusable or testable so my opinion is this change should be reverted. It would be of value if AzureLucene directory could know nothing about the azure sdk, then it could be reused.

Reply:
@nzdev will look into that, and will move initiation to factories then, to pass as paramater.   I will look how much of API we can move out from directories to helper :)

Reply:
@nzdev I find out we actually doing same thing in AzureIndexInput, I changed that to use same method, I am trying to consolidate all azure api calls now into helper :), as it can later give options to easier swap all api to other clouds storage systems.

Reply:
One way might be to create a new interface, IRemoteDirectory that doesn't reference the azure sdk.
Have the AzureHelper become AzureBlobRemoteDirectory to implement azure specific logic and refactor all the azure logic out of AzureLuceneDirectory into this.
Inject IRemoteDirectory into AzureLucenDirectory from the directory factory.
Rename AzureLuceneDirectory to RemoteSyncDirectory.

Reply:
@nzdev that was my Idea :) as we have also that branch: https://github.com/Shazwazza/Examine/pull/146 which will finish extremely simplify by abstracting out apis :)